docs/modules/ROOT/pages/explanations/decisions/postgres-monitoring.adoc

```diff
@@ -0,0 +1,65 @@
+= PostgreSQL by VSHN Availability check
+
+== Problem
+
+We need to monitor availability of PostgreSQL by VSHN to be able to measure "Up" https://products.docs.vshn.ch/products/appcat/postgresql.html#_service_level_indicator_sli[SLI]
+
+== Goals
+
+* measure Up SLI
+* idea how to integrate SLI https://products.docs.vshn.ch/products/service_levels.html#_exceptions_to_availability_guarantee[Exceptions]
+
+== Non Goals
+
+1. Collect any futher metrics
+2. SLO Definition
+3. Detailed concept of the SLI exceptions
+
+== Solutions
+
+=== Postgresql_exporter by Stackgres
+    
+    Stackgres provides us built-in postgresql_exporter running as a side car in database pod, which contains many useful metrics, but one especially usefull from availability monitoring perspective - pg_up. Pg_up is a metric created when all preconfigured checks SQL queries are properly executed and no one returned an error. Such metric could generate false positives in situation when one of the queries will be dropped by PostgreSQL. This exporter also connects locally, which makes such solution unreliable for us, because our customers connects in different way, connection has to pass through pg bouncer, it has to reuse user credentials and shouldn't generate false positives if one of two pods crash. It doesn't check for WRITE operations, only read operations.
```

Comment:
```suggestion
Stackgres provides us built-in https://github.com/prometheus-community/postgres_exporter[PostgreSQL Server Exporter] that runs as a side car, which exposes multiple useful metrics.
One especially useful from availability monitoring perspective is `pg_up`. The `pg_up` is a metric is set to `1` if all preconfigured SQL query checks are properly executed and no one returned an error. We could use this metric more or less directly as our "Up" SLI.

One limitation if this metric is that it could generate false positives in situation when one of the queries will be dropped by PostgreSQL. 
The exporter also connects locally, which makes such solution unreliable for us, because our customers connects in different way, connection has to pass through pg bouncer, it has to reuse user credentials and shouldn't generate false positives if one of two pods crash. 
It doesn't check for WRITE operations, only read operations.
```